homeassistant/components/sensor/yr.py

```diff
@@ -124,7 +125,9 @@ def update(self):
         if self._update is not None and now <= self._update:
             return
 
-        self._weather.update()
+        # Where do we schedule this update? should it be called by this update,
+        # and make this a coroutine itself, or be schedued on a times?
+        self._weather.update()  #
```

Comment:
When writing asyncio code, you have to always be aware: where am I running. In this case, this piece of code is run outside of the event loop. So to get this on the event loop, you would either run `run_coroutine_threadsafe` (if you care about the result) or `fire_coroutine_threadsafe` (when you don't care about the result.

If the method is not a coroutine but it is still Python that you want to execute inside the event loop from the outside, run `run_callback_threadsafe`. Browse `core.py` for examples.

If you're running inside the loop, you can either `yield from` another async task or you can call some of the async friendly methods (they start with `async_`).

Calling any of the non-async methods in the core will use `run_callback_threadsafe` to execute it inside the loop and return result to the thread.